perma_web/perma/utils.py

```diff
@@ -562,89 +562,88 @@ def stream_warc_if_permissible(link, user):
 #
 # Webrecorder Helpers
 #
-
-def get_wr_session(request):
-    """
-    Initializes Webrecorder session or retrieves existing session.
-    """
-    wr_username = request.session.get('wr_username')
-    wr_session_cookie = request.session.get('wr_session_cookie')
-    is_new_session = False
+def ensure_session(request, session_key, user_key, login_info):
+    session_cookie = request.session.get(session_key, '')
 
     response, data = query_wr_api(
         method='post',
-        path='/auth/anon_user',
-        cookie=wr_session_cookie,
+        path='/auth/ensure_login',
+        cookie=session_cookie,
+        json=login_info,
         valid_if=lambda code, data: code == 200
     )
 
-    # If this is a new session, WR returns a new username and session cookie:
-    # Set/Update Perma's records.
-    new_username = data['user']['username']
-    new_cookie = response.cookies.get('__wr_sesh') # NB cookie only present in api response if session is new
-    if wr_username != new_username:
-        request.session['wr_username'] = new_username
-        wr_username = new_username
-        is_new_session = True
-    if new_cookie and wr_session_cookie != new_cookie:
-        request.session['wr_session_cookie'] = new_cookie
-        wr_session_cookie = new_cookie
-        is_new_session = True
+    new_session_cookie = response.cookies.get('__wr_sesh')
+    if new_session_cookie:
+        session_cookie = new_session_cookie
+
+    request.session[session_key] = session_cookie
+    if user_key:
+        request.session[user_key] = data.get('username')
+
+    return data.get('username'), session_cookie, data.get('coll_empty')
```

Comment:
Looking at WR's `ensure_login` function, it seems like unless there's an error (which we should handle in the validation line), it should always return:
```
        result = {'username': user.name,
                  'coll_empty': coll_empty,
                  'coll_created': coll_created
                 }
```
If I'm reading that correctly, let's go ahead and access `username` and `coll_empty` keys by name here, so that we get an error if the api changes, rather than passing along `None`s to the caller.

Reply:
If I'm reading `ensure_login` correctly, it looks like it's valid if http 200 and if `'error' not in data` (since `            if not self.access.is_anon(): return {'error': 'already_logged_in'}` and `if 'success' not in result: return result`. Does that sound right to you?